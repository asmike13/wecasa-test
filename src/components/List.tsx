const List = () => {
	return (
		<div>
			<h1>Liste</h1>
		</div>
	)
}

export default List
