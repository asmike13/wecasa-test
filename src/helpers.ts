export const timeFormat = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes % 60

	const displayHours = hours > 0 ? `${hours}H` : ''
	const displayMinutes = minutes > 0 ? `${minutes}min` : ''

	return `${displayHours} ${displayMinutes}`
}

export const priceFormat = (priceInCents: number) => {
	const price = Math.floor(priceInCents / 100)
	const cents = priceInCents % 100

	const displayPrice = price > 0 ? price : ''
	const displayCents = cents > 0 ? `.${cents}` : ''

	return `${displayPrice}${displayCents}â‚¬`
}
