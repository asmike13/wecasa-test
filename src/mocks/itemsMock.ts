export const itemMock = [{
	key: 'pReference-cReference',
	prestation: {
		reference: 'pReference',
		title: 'pTitle',
		duration: 10,
		price: 10,
		quantity: 10,
	},
	category: {
		reference: 'cReference',
		title: 'cTitle',
	},
}, {
	key: 'pReference2-cReference2',
	prestation: {
		reference: 'pReference2',
		title: 'pTitle2',
		duration: 103,
		price: 13,
		quantity: 3,
	},
	category: {
		reference: 'cReference2',
		title: 'cTitle2',
	},
}]
