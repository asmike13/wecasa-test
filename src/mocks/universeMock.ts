export const universeMock = {
	reference: 'haircut',
	title: 'Coiffure',
	categories: [{
		reference: 'man',
		title: 'Homme',
		prestations: [{
			reference: 'man_shampoo',
			title: 'Shampoing',
			duration: 10,
			price: 400,
		},
		{
			reference: 'man_haircut',
			title: 'Coupe homme',
			duration: 30,
			price: 2990,
		},
		{
			reference: 'man_color',
			title: 'Couleur',
			duration: 45,
			price: 6090,
		},
		{
			reference: 'beard_trim',
			title: 'Taille de la barbe',
			duration: 15,
			price: 2000,
		}],
	},
	{
		reference: 'woman',
		title: 'Femme',
		prestations: [{
			reference: 'woman_shampoo',
			title: 'Shampoing',
			duration: 10,
			price: 400,
		},
		{
			reference: 'woman_haircare',
			title: 'Soin profond',
			duration: 10,
			price: 1000,
		},
		{
			reference: 'woman_haircut',
			title: 'Coupe cheveux courts et mi-longs',
			duration: 30,
			price: 3290,
		},
		{
			reference: 'creative_haircut',
			title: 'Coupe cheveux longs',
			duration: 45,
			price: 4290,
		},
		{
			reference: 'brushing',
			title: 'Brushing (courts et mi-longs)',
			duration: 20,
			price: 3290,
		},
		{
			reference: 'brushing_long_hair',
			title: 'Brushing (longs)',
			duration: 40,
			price: 4290,
		},
		{
			reference: 'woman_color',
			title: 'Couleur (racines)',
			duration: 45,
			price: 6090,
		},
		{
			reference: 'woman_fullcolor',
			title: 'Couleur complète',
			duration: 75,
			price: 8090,
		},
		{
			reference: 'tie_and_dye',
			title: 'Ombré hair',
			duration: 90,
			price: 9090,
		},
		{
			reference: 'woman_streaks',
			title: 'Mèches',
			duration: 120,
			price: 8090,
		},
		{
			reference: 'woman_hair_highlighting',
			title: 'Balayage',
			duration: 120,
			price: 8090,
		},
		{
			reference: 'woman_hair_permanent',
			title: 'Permanente',
			duration: 120,
			price: 8090,
		},
		{
			reference: 'haircare_patina',
			title: 'Patine',
			duration: 15,
			price: 1990,
		},
		{
			reference: 'complex_brushing',
			title: 'Brushing élaboré',
			duration: 40,
			price: 4290,
		},
		{
			reference: 'evening_brushing',
			title: 'Coiffure de soirée (chignon, tresses, attaches…) ',
			duration: 45,
			price: 5990,
		},
		{
			reference: 'sophisticated_hairdressing',
			title: 'Coiffure de mariée',
			duration: 90,
			price: 9900,
		},
		{
			reference: 'hairdressing_bride_with_test',
			title: 'Coiffure de mariage avec essai',
			duration: 90,
			price: 21900,
		},
		{
			reference: 'brazilian_smoothing',
			title: 'Lissage brésilien',
			duration: 180,
			price: 21900,
		}],
	},
	{
		reference: 'child',
		title: 'Enfant',
		prestations: [{
			reference: 'child_haircut',
			title: 'Coupe enfant',
			duration: 20,
			price: 1790,
		},
		{
			reference: 'child_hairdressing',
			title: 'Coiffure enfant (chignon, tresse, attache...) ',
			duration: 30,
			price: 3990,
		}],
	}],
}
