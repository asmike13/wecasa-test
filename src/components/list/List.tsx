import React from 'react'
import { ICategory, IPrestation, IUniverseList } from 'src/apis/universeApi'
import { priceFormat, timeFormat } from 'src/helpers/helpers'

import './styles.scss'

export interface IListProps {
	universeList: IUniverseList
	addToCart: (category: ICategory, prestation: IPrestation) => void
}

const List = ({ universeList, addToCart }: IListProps) => {
	const [tab, setTab] = React.useState(0)

	return (
		<div className='list-container'>
			<h2>{`Liste ${universeList.title}`}</h2>
			{universeList.categories.map((category, index) => (
				<button
					key={`${category.reference}-tab-btn`}
					className={index === tab ? 'selected-tab' : 'tab'}
					onClick={() => setTab(index)}
				>
					{category.title}
				</button>
			))}
			{universeList.categories.map((category, index) => (
				category.prestations.map((prestation) => (
					<div key={`${category.reference}-${prestation.reference}`} className={`prestation ${tab !== index ? 'hide' : ''}`}>
						<button onClick={() => addToCart(category, prestation)}>Add to cart</button>
						<ul>
							<li>{prestation.title}</li>
							<li>{timeFormat(prestation.duration)}</li>
							<li>{priceFormat(prestation.price)}</li>
						</ul>
					</div>
				))
			))}
		</div >
	)
}

export default List
