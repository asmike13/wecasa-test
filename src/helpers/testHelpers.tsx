import { Provider } from 'react-redux'
import { store } from 'src/store'

export const StoreWrapper = (props: any) => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
