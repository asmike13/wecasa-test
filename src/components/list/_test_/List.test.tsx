import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { StoreWrapper } from 'src/helpers/testHelpers';
import { universeMock } from 'src/mocks/universeMock';
import List, { IListProps } from '../List';

export const props: IListProps = {
	universeList: universeMock,
	addToCart: jest.fn(),
}

it('should render List as snapshot with universeMock', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<List {...props} />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})

it('should call addToCart on click', async () => {
	render(
		<StoreWrapper>
			<List {...props} />
		</StoreWrapper>,
	)

	fireEvent.click(screen.getAllByText('Add to cart')[0])
	expect(props.addToCart).toHaveBeenCalledTimes(1)
})
