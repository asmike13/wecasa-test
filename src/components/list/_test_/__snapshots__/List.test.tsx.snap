// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render List as snapshot with universeMock 1`] = `
<div
  className="list-container"
>
  <h2>
    Liste Coiffure
  </h2>
  <button
    className="selected-tab"
    onClick={[Function]}
  >
    Homme
  </button>
  <button
    className="tab"
    onClick={[Function]}
  >
    Femme
  </button>
  <button
    className="tab"
    onClick={[Function]}
  >
    Enfant
  </button>
  <div
    className="prestation "
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Shampoing
      </li>
      <li>
         10min
      </li>
      <li>
        4€
      </li>
    </ul>
  </div>
  <div
    className="prestation "
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coupe homme
      </li>
      <li>
         30min
      </li>
      <li>
        29.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation "
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Couleur
      </li>
      <li>
         45min
      </li>
      <li>
        60.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation "
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Taille de la barbe
      </li>
      <li>
         15min
      </li>
      <li>
        20€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Shampoing
      </li>
      <li>
         10min
      </li>
      <li>
        4€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Soin profond
      </li>
      <li>
         10min
      </li>
      <li>
        10€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coupe cheveux courts et mi-longs
      </li>
      <li>
         30min
      </li>
      <li>
        32.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coupe cheveux longs
      </li>
      <li>
         45min
      </li>
      <li>
        42.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Brushing (courts et mi-longs)
      </li>
      <li>
         20min
      </li>
      <li>
        32.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Brushing (longs)
      </li>
      <li>
         40min
      </li>
      <li>
        42.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Couleur (racines)
      </li>
      <li>
         45min
      </li>
      <li>
        60.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Couleur complète
      </li>
      <li>
        1H 15min
      </li>
      <li>
        80.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Ombré hair
      </li>
      <li>
        1H 30min
      </li>
      <li>
        90.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Mèches
      </li>
      <li>
        2H
      </li>
      <li>
        80.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Balayage
      </li>
      <li>
        2H
      </li>
      <li>
        80.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Permanente
      </li>
      <li>
        2H
      </li>
      <li>
        80.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Patine
      </li>
      <li>
         15min
      </li>
      <li>
        19.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Brushing élaboré
      </li>
      <li>
         40min
      </li>
      <li>
        42.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coiffure de soirée (chignon, tresses, attaches…) 
      </li>
      <li>
         45min
      </li>
      <li>
        59.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coiffure de mariée
      </li>
      <li>
        1H 30min
      </li>
      <li>
        99€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coiffure de mariage avec essai
      </li>
      <li>
        1H 30min
      </li>
      <li>
        219€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Lissage brésilien
      </li>
      <li>
        3H
      </li>
      <li>
        219€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coupe enfant
      </li>
      <li>
         20min
      </li>
      <li>
        17.90€
      </li>
    </ul>
  </div>
  <div
    className="prestation hide"
  >
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
    <ul>
      <li>
        Coiffure enfant (chignon, tresse, attache...) 
      </li>
      <li>
         30min
      </li>
      <li>
        39.90€
      </li>
    </ul>
  </div>
</div>
`;
